File: javascript/grains/grains.js
//
// This is only a SKELETON file for the 'Grains' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export class Grains {
  square(num) {
    const sq = 2**(num-1)
    return BigInt(sq).toString()
  }

  total() {
    const c64 = 2**64
    const c1 = 2**0 
    return (BigInt(c64) - BigInt(c1)).toString()
  }
}

File: ruby/grains/grains.rb
# frozen_string_literal: true

# Exercism grains
class Grains
  def self.square(num)
    raise ArgumentError if num <= 0 || num > 64

    2 << (num - 2)
  end

  def self.total
    (2 << 63) - 1
  end
end
